hardware/raspberrypi/booteeprom.md

```diff
@@ -2,82 +2,118 @@
+The images under `/lib/firmware/raspberrypi/bootloader` are part of the `rpi-eeprom` package and are only updated via `apt update`.
 
 ```
-# Disable
-sudo systemctl mask rpi-eeprom-update
+sudo rpi-eeprom-update
+```
 
-# Enable it again
-sudo systemctl unmask rpi-eeprom-update
+If an update is available, you can install it using:
+```
+sudo rpi-eeprom-update -a
+sudo reboot
 ```
 
-The `FREEZE_VERSION` option in the EEPROM config file may be used to indicate that the EEPROM should not be updated on this board. 
+### Reading current the EEPROM version
+```
+vcgencmd bootloader_version
+```
 
-Note that by default, updating the bootloader (automatically or manually) will retain any custom configuration options of the previous installed version. You can override the migration by manually updating and using the `-d` option with `rpi-eeprom-update`. This will force the updated bootloader to use its inbuilt defaults.
+## Updating the EEPROM configuration
+The bootloader EEPROM image contains a configuration file to define the boot behaviour (e.g. selecting between SD, network and USB boot). 
+The `rpi-eeprom-config` tool may be used to modify the configuration within an EEPROM image file. 
 
-## Manually checking if an update is available
+See the [Bootloader Configuration Page](./bcm2711_bootloader_config.md) for details of the configuration file.
 
-Running the `rpi-eeprom-update` command with no parameters indicates whether an update is required. An update is required if the timestamp of the most recent file in the firmware directory (normally `/lib/firmware/raspberrypi/bootloader/critical`) is newer than that reported by the current bootloader.
-The images under `/lib/firmware/raspberrypi/bootloader` are part of the `rpi-eeprom` package and are only updated via `apt update`.
+### Reading the current EEPROM configuration
+To view the configuration file used by the bootloader at boot time run `rpi-eeprom-config` or `vcgencmd bootloader_config`.
+
+### Editing the current bootloader configuration
+The following command loads the current EEPROM configuration into a text editor. When the editor is closed, `rpi-eeprom-config` applies the updated configuration to latest available EEPROM release and uses `rpi-eeprom-update` to schedule an update when the system is rebooted:
 
 ```
-sudo rpi-eeprom-update
+sudo -E rpi-eeprom-config --edit
+sudo reboot
 ```
+If the updated configuration file is identical or empty then no changes are made.
 
-If an update is available, you can install it using :
+The editor is selected by the `EDITOR` environment variable.
 
+### Applying a saved configuration file
+The following command applies `boot.conf` to the latest available EEPROM image and uses `rpi-eeprom-update` to schedule an update when the system is rebooted.
 ```
-sudo rpi-eeprom-update -a
+sudo rpi-eeprom-config --apply boot.conf
 sudo reboot
 ```
 
-### Reading the current EEPROM configuration
+### Reading the configuration file from an EEPROM image
```

Comment:
Might be worth swapping things around so that "### Reading the configuration file from an EEPROM image" comes before "### Applying a saved configuration file" ?

Reply:
good point. done